numba/runtime/_nrt_python.c

```diff
@@ -156,7 +168,7 @@ int MemInfo_init(MemInfoObject *self, PyObject *args, PyObject *kwds) {
     if(PyErr_Occurred()) return -1;
     self->meminfo = (MemInfo*)raw_ptr;
     self->defer = 0;
-    NRT_MemInfo_acquire(self->meminfo);
+    assert (NRT_MemInfo_refcount(self->meminfo) > 0 && "0 refcount");
```

Comment:
It doesn't sound very valuable to have asserts if nobody compiles in debug mode :-)